src/git.rs

```diff
@@ -13,29 +13,35 @@ pub fn git_fetch(root: &Path) -> Result<(), git2::Error> {
 }
 
 #[derive(Serialize, Deserialize)]
+pub struct BranchResponse {
+    default_branch: Option<String>,
+    branches: Vec<BranchInfo>,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
 pub struct BranchInfo {
     name: String,
     commit_hash: String,
     time: i64,
     message: Vec<String>,
 }
 
-fn get_branch_info(
-    branch_struct: Result<(Branch, BranchType), git2::Error>,
-) -> Result<BranchInfo, git2::Error> {
-    let branch = branch_struct?.0;
-    let lossy_name = String::from_utf8_lossy(branch.name_bytes()?);
-    let name = lossy_name
-        .strip_prefix("origin/")
-        .unwrap_or(&lossy_name)
-        .to_owned();
-
-    let branch_commit = branch.into_reference().peel_to_commit()?;
+fn get_branch_name(reference: &Reference) -> String {
+    let lossy_name = String::from_utf8_lossy(reference.name_bytes());
+    lossy_name
+        .strip_prefix("refs/remotes/origin/")
+        .unwrap_or_else(|| &lossy_name)
+        .to_string()
+}
+
+fn get_branch_info(branch: Branch) -> Result<BranchInfo, git2::Error> {
+    let git_ref = branch.get().resolve()?;
+    let name = get_branch_name(&git_ref);
```

Comment:
Is this still needed? Seems like that was used when you were calling `get_branch_info` on `origin/HEAD`, but now that you don't anymore you can just keep the old code and do `branch.name_bytes()`.